src/core/net/ip6_address.cpp

```diff
@@ -89,6 +89,11 @@ bool Address::IsRealmLocalMulticast(void) const
     return IsMulticast() && (GetScope() == kRealmLocalScope);
 }
 
+bool Address::IsMulticastHigherThanRealmLocal(void) const
+{
+    return (mFields.m8[0] == 0xff) && (GetScope() > kRealmLocalScope);
```

Comment:
nit: for consistency with above methods, can we use `IsMulticast()` instead of `mFields.m8[0] == 0xff`